dump/main.c

```diff
@@ -140,9 +140,68 @@ static int dump_full(const char* spec, bool used_sectors)
 	return 0;
 }
 
+static int dump_file(const char* spec, const char* path)
+{
+	struct exfat ef;
+	struct exfat_node* node;
+	cluster_t cluster, next_cluster;
+	cluster_t start_cluster = 0;
+	off_t remainder;
+	off_t extent_size;
+	off_t lsize;
+	int rc;
+
+	if (exfat_mount(&ef, spec, "ro") != 0)
+		return 1;
+
+	rc = exfat_lookup(&ef, &node, path);
```

Comment:
You should check the return value.

Also you should call exfat_put_node() after each successful exfat_lookup().